src/linear_algebra/matrix.lean

```diff
@@ -232,7 +232,7 @@ by rw [matrix.to_lin'_apply, hv₂.equiv_fun_symm_apply]
 by simp only [matrix.to_lin_apply, matrix.mul_vec, dot_product, hv₁.equiv_fun_self, mul_boole,
   finset.sum_ite_eq, finset.mem_univ, if_true]
 
-@[simp]
+/-- This will be a special case of `linear_map.to_matrix_id_eq_basis_to_matrix`. -/
 lemma linear_map.to_matrix_id : linear_map.to_matrix hv₁ hv₁ id = 1 :=
```

Comment:
Why is this a bad simp lemma?

Reply:
The `simp_nf` linter complains because it can be replaced with `by [linear_map.to_matrix_id_eq_basis_to_matrix, is_basis.to_matrix_self]`.